nixos/modules/services/security/authelia.nix

```
@@ -8,7 +8,6 @@ let
   cfg = config.services.authelia;
 
   format = pkgs.formats.yaml { };
-  configFile = format.generate "config.yml" cfg.settings;
 
   autheliaOpts = with lib; { name, ... }: {
```

```suggestion
  autheliaOpts = with lib; { name, config, ... }: {
```
This `config` will be scoped to per-instance, because `autheliaOpts` is passed to `types.submodule`.

That way, it should be possible to do warnings per affected instance.

---

Using `config` here allows for accessing the instance configuration (thank you for the tip), however I still get the infinite recursion :/

I think that it is because I try to lazily define `settings` (i.e. the `address` option of the submodule `settings`) while trying to access the "final state" of `settings` (i.e. the `host`/`port`/`path`).

When I set the condition to something outside `settings` (e.g. `config.enable`), it works.
It is when I try to use something in `settings` that it breaks.

Also, when using eager functions (e.g. `lib.optionalString`), it compiles, but the key is still defined in the YAML configuration file (`address: ''`), and Authelia yells about defining both old and new keys.

---

Hm :/ I'd have to take a look to say more, but I guess last resort would be a top-level `mapAttrsToList`, to generate `config.warnings` manually ....